bluesky/run_engine.py

```diff
@@ -288,6 +359,7 @@ def __init__(self, md=None, *, loop=None, preprocessors=None,
         self._exit_status = 'success'  # optimistic default
         self._reason = ''  # reason for abort
         self._task = None  # asyncio.Task associated with call to self._run
+        self._task_fut = None  # asyncio.Task associated with call to self._run
```

Comment:
This comment and the comment in the line above it are identical.

Reply:
ah, this was confusing.  We used to get the task back directly from the asyncio call, but now we get back a future that knows about the task and acts as a proxy.  When you call `tsk.cancel()` it is left in a "please cancel the next time you get a chance" state (as it arranges to throw the `CancelledError` down into the wrapped co-ro).  However, when you call `future.cancel()` the proxy forwards cancel call and uncontitionally marks itself as cancelled.  Meanwhile, the underlying task may have delayed (or entirely eaten as we do) the cancellation and still be a going concern.   This effectively de-couples the future from the task which breaks things.

I may report this as a bug upstream, but need to write a ah, more minimal, example of what is going on.